sqlmesh/core/context.py

```diff
@@ -422,11 +425,21 @@ def run(
                 NotificationEvent.RUN_FAILURE, traceback.format_exc()
             )
             raise e
-        self.notification_target_manager.notify(NotificationEvent.RUN_END, environment=environment)
+        if success:
+            self.notification_target_manager.notify(
+                NotificationEvent.RUN_END, environment=environment
+            )
+        else:
+            self.notification_target_manager.notify(
+                NotificationEvent.RUN_FAILURE, environment=environment
+            )
+            return success
```

Comment:
I currently have it not run janitor if the run had an error. Let me know if anyone disagrees with that.